examples/experimental/dagster-airlift/dagster_airlift/core/airflow_cacheable_assets_def.py

```diff
@@ -0,0 +1,421 @@
+                tags={
+                    **spec.tags,
+                    MIGRATED_TAG: str(migration_state_for_task),
+                    DAG_ID_TAG: task_info.dag_id,
+                    TASK_ID_TAG: task_info.task_id,
+                },
+                deps=spec_deps,
+                group_name=spec.group_name,
+            )
+            all_asset_keys_per_dag_id[task_info.dag_id].add(spec.key)
+    return (
+        cacheable_specs_per_asset_key,
+        all_asset_keys_per_dag_id,
+        downstreams_asset_dependency_graph,
+    )
+
+
+def construct_assets_with_task_migration_info_applied(
+    definitions: Optional[Definitions],
+    cacheable_specs: CacheableSpecs,
+) -> List[AssetsDefinition]:
+    if not definitions or not definitions.assets:
+        return []
+
+    new_assets_defs = []
+    for asset in definitions.assets:
+        assets_def = check.inst(
+            asset, AssetsDefinition, "Expected orchestrated defs to all be AssetsDefinitions."
+        )
+        overall_migration_status = None
+        overall_task_id = None
+        overall_dag_id = None
+        new_specs = []
+        for spec in assets_def.specs:
+            if not cacheable_specs.task_asset_specs.get(spec.key):
+                raise Exception(
+                    f"Could not find cacheable spec for asset key {spec.key.to_user_string()}"
+                )
+            new_spec = cacheable_specs.task_asset_specs[spec.key].to_asset_spec()
+            if not new_spec.tags.get(MIGRATED_TAG):
+                raise Exception(
+                    f"Could not find migrated status for asset key {spec.key.to_user_string()}"
+                )
+            check.invariant(
+                overall_migration_status is None
+                or overall_migration_status == new_spec.tags[MIGRATED_TAG],
+                "Expected all assets in an AssetsDefinition (and therefore the same task) to have the same migration status.",
+            )
+            overall_migration_status = new_spec.tags[MIGRATED_TAG]
+            dag_id = new_spec.tags.get(DAG_ID_TAG)
+            if not dag_id:
+                raise Exception(f"Could not find dag ID for asset key {spec.key.to_user_string()}")
+            check.invariant(
+                overall_dag_id is None or overall_dag_id == dag_id,
+                "Expected all assets in an AssetsDefinition to have the same dag ID.",
+            )
+            overall_dag_id = dag_id
+            task_id = new_spec.tags.get(TASK_ID_TAG)
+            if not task_id:
+                raise Exception(f"Could not find task ID for asset key {spec.key.to_user_string()}")
+            check.invariant(
+                overall_task_id is None or overall_task_id == task_id,
+                "Expected all assets in an AssetsDefinition to have the same task ID.",
+            )
+            overall_task_id = task_id
+            new_specs.append(new_spec)
+
+        if overall_migration_status == "True":
```

Comment:
seems weird to not coerce this to a bool prior to comparison

Reply:
I originally tried this but it's actually more complicated than initially anticipated: https://stackoverflow.com/questions/715417/converting-from-a-string-to-boolean-in-python (essentially bool("False") evaluates to true, and the only reliable coercion method is going away in 3.12).
I'll leave a comment explaining though.